vortex-array/src/flatten.rs

```diff
@@ -1,14 +1,32 @@
+use std::sync::Arc;
```

Comment:
Almost time to turn this into a directory. IntelliJ can right-click + promote module to directory. Then could pull out arrow logic. No strong preference

Reply:
i'm just gonna leave it for now. i feel like it's not quite hard enough to search through just yet...maybe in 100 lines or so